From: Michael Webster <miketwebster@gmail.com>
Date: Mon, 7 Nov 2022 16:48:57 -0500
Subject: tiling: Skip the resize effect for tiled windows during user grabs.

meta_window_tile gets called by the grabbed window's match during
tile resizing. These incremental changes don't need to be animated.

(cherry picked from commit 559e6ff327a358a6daa4b0683fee95700591749f)

Bug: https://gitlab.gnome.org/GNOME/mutter/-/issues/2246
Forwarded: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2579>
Origin: upstream, 43.3, commit:82eb4cdbae909798fdc08318663a4e83806e580d
---
 src/core/window.c | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/src/core/window.c b/src/core/window.c
index 744e736..7e1c35c 100644
--- a/src/core/window.c
+++ b/src/core/window.c
@@ -2910,7 +2910,6 @@ meta_window_tile (MetaWindow   *window,
                   MetaTileMode  tile_mode)
 {
   MetaMaximizeFlags directions;
-  MetaRectangle old_frame_rect, old_buffer_rect;
 
   g_return_if_fail (META_IS_WINDOW (window));
 
@@ -2939,12 +2938,17 @@ meta_window_tile (MetaWindow   *window,
   /* Setup the edge constraints */
   update_edge_constraints (window);
 
-  meta_window_get_frame_rect (window, &old_frame_rect);
-  meta_window_get_buffer_rect (window, &old_buffer_rect);
+  if (!window->tile_match || window->tile_match != window->display->grab_window)
+    {
+      MetaRectangle old_frame_rect, old_buffer_rect;
+
+      meta_window_get_frame_rect (window, &old_frame_rect);
+      meta_window_get_buffer_rect (window, &old_buffer_rect);
 
-  meta_compositor_size_change_window (window->display->compositor, window,
-                                      META_SIZE_CHANGE_MAXIMIZE,
-                                      &old_frame_rect, &old_buffer_rect);
+      meta_compositor_size_change_window (window->display->compositor, window,
+                                          META_SIZE_CHANGE_MAXIMIZE,
+                                          &old_frame_rect, &old_buffer_rect);
+    }
 
   meta_window_move_resize_internal (window,
                                     (META_MOVE_RESIZE_MOVE_ACTION |
